### DISCUSSION TITLE
Possible to remove /api from url?

### DISCUSSION THREAD

**Discussion by kgn** [vercel/community#3205] [Help]
### Problem Description

Hi all!

I am working on a simple function to redirect to another website, but the logic is conditional based on what is passed in so I can't just use redirects in vercel.json. I have my code working but to get my serverless function to run I seem to have to have it in a /api folder and have /api in the name. These urls are for referral links that people will see and share so I want to keep them short and don't want to have /api in them.

Is there a way to have serverless functions run at the root directory instead of the /api directory?

**Comment by amyegan** [comment]
👋 Hi @kgn. You can use [conditional rewrites in Middleware](https://vercel.com/docs/concepts/functions/edge-middleware/middleware-api#match-paths-based-on-conditional-statements) or set up [`rewrites`](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) in a `vercel.json` file to hide `/api` from the URL

**Comment by kgn** [comment]
Hi @amyegan, thank you for this suggestion! I gave it a try but think my use case might be too complex because I'm pulling data from airtable to know which url to redirect too. Are edge functions able to call other apis like airtable? Is there an execution time limit?
